gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -599,14 +693,140 @@ def get_date_created(self, id):
 
         return date
 
+    def create(self, validated_data, disease_obj, publications_list):
+        """
+            Create a G2P record.
+            A record is always linked to one or more panels and publications.
+
+            Mandatory data:
+                            - locus
+                            - G2P stable_id
+                            - disease
+                            - genotype (allelic requeriment)
+                            - mechanism (TODO)
+                            - panel(s)
+                            - confidence
+                            - publications
+        """
+
+        locus_name = validated_data.get('locus') # Usually this is the gene symbol
+        stable_id_obj = validated_data.get('stable_id') # stable id obj
+        genotype = validated_data.get('allelic_requirement') # allelic requirement
+        panels = validated_data.get('panels') # Array of panel names
+        confidence = validated_data.get('confidence') # confidence level and justification
+
+        if not panels or not publications_list:
+            raise serializers.ValidationError({"message": f"Missing data to create the G2P record {stable_id_obj.stable_id}"})
+
+        # Check if record (LGD) is already inserted
+        try:
+            lgd_obj = LocusGenotypeDisease.objects.get(stable_id=stable_id_obj)
+            return lgd_obj
+
+        except LocusGenotypeDisease.DoesNotExist:
+
+            # Get locus object
+            try:
+                locus_obj = Locus.objects.get(name=locus_name)
+            except Locus.DoesNotExist:
+                raise serializers.ValidationError({"message": f"Invalid locus {locus_name}"})
+
+            # Get genotype
+            try:
+                genotype_obj = Attrib.objects.get(
+                    value = genotype,
+                    type__code = "genotype"
+                )
+            except Attrib.DoesNotExist:
+                raise serializers.ValidationError({"message": f"Invalid genotype value {genotype}"})
+
+            # Get confidence
+            try:
+                confidence_obj = Attrib.objects.get(
+                    value = confidence["level"],
+                    type__code = "confidence_category"
+                )
+            except Attrib.DoesNotExist:
+                raise serializers.ValidationError({"message": f"Invalid confidence value {confidence["level"]}"})
```

Comment:
Running this in my local is giving the error "SyntaxError: f-string: unmatched '['"
After changing "level"(surrounded by double quotes) to 'level' (surrounded by single quotes) fixed the issue
Please check if you are facing similar issue in your local

Reply:
You're right, the correct syntax is `{confidence['level']}`